packages/gatsby/src/components/hooks.tsx

```diff
@@ -0,0 +1,35 @@
+import {StaticQuery, graphql} from 'gatsby';
+import React                  from 'react';
+
+export function Hooks() {
+  return (
+    <StaticQuery
+      query={graphql`
+        query MyQuery {
+          yarnIntrospection(id: {eq: "0f47090c-afdf-5a02-98f8-047adfe6297f"}) {
+            value {
+              name
+              definition
+              comment
+              file
+            }
+          }
+        }
+      `}
+      render={data => <>
+        {data.yarnIntrospection.value.map(({name, comment, definition}) => (
+          <div key={name}>
+            <h3 id={`hook-${name}`}><code>{name}</code></h3>
+            {comment.split(/\n\n/).map((text, index) => (
+              <p key={index}>{text}</p>
+            ))}
+            <pre><code>
+              {definition}
+            </code></pre>
```

Comment:
TS syntax highlighting would be useful here.

Reply:
Same issue; the syntax highlight is powered by markdown, which doesn't run within the react component. Only option I see would be to duplicate the highlight pipeline :/

Reply:
I'll leave them unstyled for now; should we revamp the website someday, I'd like to set it up from the beginning to support Shiki Twoslash

Reply:
I _think_ shiki twoslash can work in mdx files in Gatsby mostly OOTB via the remark plugin, I used that for a next.js site ( also your pnp fix has shipped, thanks! )